sw/device/lib/runtime/ibex.h

```diff
@@ -20,22 +20,29 @@
 extern const size_t kIbexClockFreqHz;
 
 /**
- * Spins for roughly the number of |cycles| given. For best results, |cycles|
- * should be a multiple of eight.
+ * Read the cycle counter.
  *
- * This function should not be used for time-keeping.
+ * The value of the counter is stored across two 32-bit registers: |mcycle| and
+ * |mcycleh|. This function is guaranteed to return a valid 64-bit cycle
+ * counter value, even if |mcycle| overflows before reading |mcycleh|.
  *
- * @param cycles the number of cycles to burn.
+ * Adapted from: The RISC-V Instruction Set Manual, Volume I: Unprivileged ISA
+ * V20191213, pp. 61.
  */
-inline void ibex_busy_loop(size_t cycles) {
-  size_t out;  // Used to create an inout parameter below.
+inline uint64_t read_cycle_counter() {
```

Comment:
- This isn't really Ibex-specific, so it probably can go into another file if you'd like. But fine with me either way.
- I'd also think about a name which sounds a bit more like a getter, like `get_mcycle()`. But again, up to you.

Reply:
Renamed to `ibex_mcycle_read()`.